src/saltext/vmware/modules/esxi.py

```diff
@@ -698,6 +698,210 @@ def get_firewall_config(
+
+    firewall_config
+        Dict of Rule set to be used to change Firewall configuration. Eg: {"name": "CIMHttpServer"}
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_esxi.set_firewall_config
+    """
+    log.debug("Running vmware_esxi.set_firewall_config")
+    ret = []
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=host_name is None,
+    )
+    try:
+        for host in hosts:
+            firewall = host.configManager.firewallSystem
+            if not firewall:
+                continue
+            firewall_rulespec = vim.host.Ruleset.RulesetSpec()
+            firewall_rulespec.allowedHosts = vim.host.Ruleset.IpList()
+            if "enabled" in firewall_config and firewall_config["enabled"]:
+                firewall.EnableRuleset(id=firewall_config["name"])
+            else:
+                firewall.DisableRuleset(id=firewall_config["name"])
+            if "allowed_host" in firewall_config:
+                if "all_ip" in firewall_config["allowed_host"]:
+                    firewall_rulespec.allowedHosts.allIp = firewall_config["allowed_host"]["all_ip"]
+                if "ip_address" in firewall_config["allowed_host"]:
+                    firewall_rulespec.allowedHosts.ipAddress = list(
+                        firewall_config["allowed_host"]["ip_address"]
+                    )
+                firewall_rulespec.allowedHosts.ipNetwork = []
+                if "ip_network" in firewall_config["allowed_host"]:
+                    for network in firewall_config["allowed_host"]["ip_network"]:
+                        address, mask = network.split("/")
+                        tmp_ip_network_spec = vim.host.Ruleset.IpNetwork()
+                        tmp_ip_network_spec.network = address
+                        tmp_ip_network_spec.prefixLength = int(mask)
+                        firewall_rulespec.allowedHosts.ipNetwork.append(tmp_ip_network_spec)
+                firewall.UpdateRuleset(id=firewall_config["name"], spec=firewall_rulespec)
+            res = get_firewall_config(
+                firewall_config["name"], host_name=host.name, service_instance=service_instance
+            )
+            ret.append(res)
+        return ret
+
+    except DEFAULT_EXCEPTIONS as exc:
+        raise salt.exceptions.SaltException(str(exc))
+
+
+def set_firewall_configs(
```

Comment:
similar compliant to get_firewall_config and get_firewall_configs above. far too easy to mistype or mis-think the function desired - change the function names to have a clearer name which will prevent misuse or mistake.